internal/proxy/infrastructure/cache/lru_blockchain_repository.go

```diff
@@ -39,3 +39,7 @@ func (l lruBlockchainRepository) Add(request *pkgmodel.Request, response interfa
 
 	return nil
 }
+
+func (p lruBlockchainRepository) IsRollingRedirection(url string) bool {
```

Comment:
IsRollingRedirection is more a business logic than an interaction with the blockchain repository, so I suggest to move this method to the proxy use case